File: programmers/js/basic/181836.js
// 그림 확대
function solution(picture, k) {
  return picture.reduce((acc, cur) => {
    for (let i = 0; i < k; i++) {
      acc.push([...cur].map((px) => px.repeat(k)).join(""));
    }

    return acc;
  }, []);
}

File: programmers/java/basic/181836.java
// 그림 확대

import java.util.ArrayList;
import java.util.Arrays;
import java.util.stream.Collectors;

class Solution {
    public String[] solution(String[] picture, int k) {
        ArrayList<String> result = new ArrayList<>();
        
        for (String row : picture) {
            for (int i = 0; i < k; i++) {
                result.add(Arrays.stream(row.split(""))
                    .map(px -> px.repeat(k))
                    .collect(Collectors.joining()));
            }
        }
        
        return result.stream().toArray(String[]::new);
    }
}